From 6425bc97bde1b8f02a8eb4c02155c715a7ac7330 Mon Sep 17 00:00:00 2001
From: "Li, HaihongX" <haihongx.li@intel.com>
Date: Fri, 1 Apr 2022 11:16:07 +0800
Subject: [PATCH] RBC(Render Buffer Compression) Support

Set bo->prime_fds[1] = handle->data[0] for rbc supporting
if modifier is I915_FORMAT_MOD_Y_TILED_CCSi or
I915_FORMAT_MOD_Y_TILED_GEN12_RC_CCS.

Tracked-On: OAM-101691
Signed-off-by: Li, HaihongX <haihongx.li@intel.com>
---
 bufferinfo/BufferInfoMapperMetadata.cpp | 6 +++++-
 include/drmhwcomposer.h                 | 1 +
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/bufferinfo/BufferInfoMapperMetadata.cpp b/bufferinfo/BufferInfoMapperMetadata.cpp
index 23a9072..a251c0d 100644
--- a/bufferinfo/BufferInfoMapperMetadata.cpp
+++ b/bufferinfo/BufferInfoMapperMetadata.cpp
@@ -51,7 +51,11 @@ BufferInfoMapperMetadata::GetFds(buffer_handle_t handle, hwc_drm_bo_t *bo) {
 
   if (num_fds >= 1 && num_fds <= 2) {
     if (IsDrmFormatRgb(bo->format)) {
-      bo->prime_fds[0] = handle->data[0];
+      if (bo->modifiers[0] == I915_FORMAT_MOD_Y_TILED_CCS ||
+          bo->modifiers[0] == I915_FORMAT_MOD_Y_TILED_GEN12_RC_CCS)
+        bo->prime_fds[0] = bo->prime_fds[1] = handle->data[0];
+      else
+        bo->prime_fds[0] = handle->data[0];
     } else {
       bo->prime_fds[0] = bo->prime_fds[1] = bo->prime_fds[2] = handle->data[0];
     }
diff --git a/include/drmhwcomposer.h b/include/drmhwcomposer.h
index 6528f86..181a7a8 100644
--- a/include/drmhwcomposer.h
+++ b/include/drmhwcomposer.h
@@ -29,6 +29,7 @@
 #include "utils/UniqueFd.h"
 
 #define DRM_FORMAT_NV12_Y_TILED_INTEL fourcc_code('9', '9', '9', '6')
+#define I915_FORMAT_MOD_Y_TILED_GEN12_RC_CCS fourcc_mod_code(INTEL, 6)
 namespace android {
 
 class DrmFbIdHandle;
-- 
2.33.1

